chain/src/main/scala/org/bitcoins/chain/blockchain/BaseBlockChain.scala

```diff
@@ -211,4 +198,25 @@ private[blockchain] trait BaseBlockChainCompObject
 
   }
 
+  /**
+    * Finds the parent of the given header
+    */
+  private def findPrevBlockHeaderIdx(
+      header: BlockHeader,
+      blockchain: Blockchain): Option[(BlockHeaderDb, Int)] = {
+    // Let's see if we are lucky and the latest tip is the parent.
+    val prevTipOpt = blockchain.headers.headOption
```

Comment:
I would prefer to use `blockchain.tip` here, in case we ever decide to organize headers a different way in `blockchain`.